src/segment_owm.go

```diff
@@ -61,22 +65,58 @@ func (d *owm) string() string {
 	return text
 }
 
-func (d *owm) setStatus() error {
+func (d *owm) getResult() (*OWMDataResponse, error) {
 	apikey := d.props.getString(APIKEY, ".")
 	location := d.props.getString(LOCATION, "De Bilt,NL")
 	units := d.props.getString(UNITS, "standard")
 	timeout := d.props.getInt(HTTPTimeout, DefaultHTTPTimeout)
+	cachefile := d.props.getString(CACHEFILE, "")
+
+	if cachefile != "" {
+		stats, err := os.Stat(cachefile)
+		if err == nil {
```

Comment:
We'll need to remove the arrow of `if`'s here. Try to return early and only keep 1 level of indentation.